tests/tests-hz/src/main/java/com/hazelcast/simulator/tests/map/ComplexSqlPredicateTest.java

```diff
@@ -0,0 +1,545 @@
+
+public class ComplexSqlPredicateTest extends HazelcastTest {
+
+    // properties
+    public int keyCount = 1000;
+
+    public int idLength = 26;
+    public int universalIdLength = 16;
+    public int panFirst6 = 999999;
+    public int panLast4 = 9999;
+    public int authCode = 999999;
+
+    private IMap<Integer, Transaction> map;
+
+    private String[] generatedIds;
+
+    @Setup
+    public void setup() {
+        this.map = targetInstance.getMap(name);
+        generatedIds = generateStrings(keyCount, idLength);
+    }
+
+    @Prepare(global = true)
+    public void prepare() {
+        Random random = new Random();
+        Streamer<Integer, Transaction> streamer = StreamerFactory.getInstance(map);
+        for (int i = 0; i < keyCount; i++) {
+            System.out.println("I: " + i);
+            Transaction value = generateTransaction(i, random);
+            streamer.pushEntry(i, value);
+        }
+        streamer.await();
+        logger.info("Map size is: " + map.size());
+    }
+
+    private Transaction generateTransaction(Integer keyId, Random random) {
+        System.out.println("GeneratedIds length: " + generatedIds.length);
+        String id = generatedIds[keyId];
+        String universalId = generateString(universalIdLength);
+        ActionCode actionCode = random.nextBoolean() ? ActionCode.CODE_0000 : ActionCode.CODE_9280;
+        String action = actionCode.getInternalCode();
+        Long amount = random.nextLong();
+        long timestamp = random.nextLong();
+        Integer panFirst6 = random.nextInt(this.panFirst6);
+        Integer panLast4 = random.nextInt(this.panLast4);
+        int possibleAuthCode = random.nextInt(this.authCode);
+        String authCode = random.nextBoolean() ? String.valueOf(possibleAuthCode) : null;
+        String acquirerId = authCode != null ? generateString(universalIdLength) : null;
+        TransactionType transactionType = TransactionType.values()[random.nextInt(TransactionType.values().length)];
+        Country country = Country.values()[random.nextInt(Country.values().length)];
+        Currency currency = Currency.values()[random.nextInt(Currency.values().length)];
+        CardBrand cardBrand = CardBrand.values()[random.nextInt(CardBrand.values().length)];
+        LineOfBusiness lineOfBusiness = LineOfBusiness.values()[random.nextInt(LineOfBusiness.values().length)];
+
+        return new Transaction(id, universalId, action, actionCode, amount, timestamp, panFirst6, panLast4, acquirerId, authCode, transactionType, country, currency, cardBrand, lineOfBusiness);
+    }
+
+    /**
+     * Should retrieve only single entry.
+     *
+     * @param state
+     */
+    @TimeStep(prob = 0.4)
+    public void searchSingle(ThreadState state) {
+        String predicate = "id = '" + generatedIds[state.random.nextInt(generatedIds.length)] + "'";
+        System.out.println( map.values(new SqlPredicate(predicate)));
```

Comment:
I hope you are going to remove system.out from the code :)

Reply:
Oh, leftover from testing :)